core/src/test/java/apoc/trigger/TriggerClusterRoutingTest.java

```diff
@@ -102,8 +105,55 @@ private static void triggerInSysLeaderMemberCommon(String query, String triggerN
         }
     }
 
+    @Test
+    public void testTriggersAllowedOnlyWithAdmin() {
+        for (Neo4jContainerExtension container: cluster.getClusterMembers()) {
+            
+            try (Session sysSession = container.getDriver().session(SessionConfig.forDatabase(SYSTEM_DATABASE_NAME))) {
+                if (!sysIsLeader(sysSession)) {
+                    return;
+                }
+                sysSession.run("CREATE USER nonadmin SET PASSWORD 'test' SET PASSWORD CHANGE NOT REQUIRED");
+            }
+
+            try (Driver userDriver = GraphDatabase.driver(container.getBoltUrl(), AuthTokens.basic("nonadmin", "test"))) {
+                
+                try (Session sysUserSession = userDriver.session(SessionConfig.forDatabase(SYSTEM_DATABASE_NAME))) {
+                    failsWithNonAdminUser(sysUserSession, "apoc.trigger.install", "call apoc.trigger.install('neo4j', 'qwe', 'return 1', {})");
+                    failsWithNonAdminUser(sysUserSession, "apoc.trigger.drop", "call apoc.trigger.drop('neo4j', 'qwe')");
+                    failsWithNonAdminUser(sysUserSession, "apoc.trigger.dropAll", "call apoc.trigger.dropAll('neo4j')");
+                    failsWithNonAdminUser(sysUserSession, "apoc.trigger.stop", "call apoc.trigger.stop('neo4j', 'qwe')");
+                    failsWithNonAdminUser(sysUserSession, "apoc.trigger.start", "call apoc.trigger.start('neo4j', 'qwe')");
+                }
+                
+                try (Session neo4jUserSession = userDriver.session(SessionConfig.forDatabase(DEFAULT_DATABASE_NAME))) {
+                    failsWithNonAdminUser(neo4jUserSession, "apoc.trigger.add", "call apoc.trigger.add('abc', 'return 1', {})");
+                    failsWithNonAdminUser(neo4jUserSession, "apoc.trigger.remove", "call apoc.trigger.remove('abc')");
+                    failsWithNonAdminUser(neo4jUserSession, "apoc.trigger.removeAll", "call apoc.trigger.removeAll");
+                    failsWithNonAdminUser(neo4jUserSession, "apoc.trigger.pause", "call apoc.trigger.pause('abc')");
+                    failsWithNonAdminUser(neo4jUserSession, "apoc.trigger.resume", "call apoc.trigger.resume('abc')");
+                }
+            }
+            
+        }
+    }
+
     private static boolean sysIsLeader(Session session) {
         final String systemRole = TestContainerUtil.singleResultFirstColumn(session, "CALL dbms.cluster.role('system')");
         return "LEADER".equals(systemRole);
     }
+    
+    private void failsWithNonAdminUser(Session session, String procName, String query) {
+        try {
+            testCall(session, query,
+                    row -> fail("Should fail because of non admin user") );
```

Comment:
```suggestion
            testCall(session, query,
                    row -> fail("Should fail because of non admin user") );
            Assertions.fail("should fail")
```